test/pylib/scylla_server.py

```diff
@@ -0,0 +1,307 @@
+truncate_request_timeout_in_ms: 300000
+write_request_timeout_in_ms: 300000
+request_timeout_in_ms: 300000
+
+# Set up authentication in order to allow testing this module
+# and other modules dependent on it: e.g. service levels
+
+authenticator: PasswordAuthenticator
+strict_allow_filtering: true
+"""
+
+# Seastar options can not be passed through scylla.yaml, use command line
+# for them. Keep everything else in the configuration file to make
+# it easier to restart. Sic: if you make a typo on the command line,
+# Scylla refuses to boot.
+SCYLLA_CMDLINE_OPTIONS = [
+    '--smp', '2',
+    '-m', '1G',
+    '--collectd', '0',
+    '--overprovisioned',
+    '--max-networking-io-control-blocks', '100',
+    '--unsafe-bypass-fsync', '1',
+    '--kernel-page-cache', '1',
+]
+
+
+class ScyllaServer:
+    # Regular expression to scan the log
+    STARTUP_MSG_RE = re.compile(".*Scylla.*initialization completed")
+
+    def __init__(self, exe=None, vardir=None, host_registry=None,
+                 cluster_name=None, seed=None, cmdline_options=None) -> None:
+        self.exe = pathlib.Path(exe).resolve()
+        self.vardir = pathlib.Path(vardir)
+        self.host_registry = host_registry
+        self.cmdline_options = cmdline_options
+        self.cfg = {
+            "cluster_name": cluster_name,
+            "host": None,
+            "seeds": seed,
+            "workdir": None,
+        }
+        self.cmd: Optional[Any] = None
+
+        async def stop_server():
+            if self.is_running:
+                await self.stop()
+
+        async def uninstall_server():
+            await self.uninstall()
+
+        self.stop_artifact = stop_server
+        self.uninstall_artifact = uninstall_server
+
+    async def install_and_start(self) -> None:
+        await self.install()
+
+        logging.info("starting server at host %s...", self.cfg["host"])
+
+        await self.start()
+
+        if self.cmd:
+            logging.info("started server at host %s, pid %d", self.cfg["host"], self.cmd.pid)
+
+    @property
+    def is_running(self) -> bool:
+        return self.cmd is not None
+
+    @property
+    def host(self) -> str:
+        return self.cfg["host"]
+
+    def find_scylla_executable(self) -> None:
+        if not os.access(self.exe, os.X_OK):
+            raise RuntimeError("{} is not executable", self.exe)
+
+    async def install(self) -> None:
+        """Create a working directory with all subdirectories, initialize
+        a configuration file."""
+
+        self.find_scylla_executable()
+
+        # Scylla assumes all instances of a cluster use the same port,
+        # so each instance needs an own IP address.
+        self.cfg["host"] = await self.host_registry.lease_host()
+        self.cfg["seeds"] = self.cfg.get("seeds") or self.cfg["host"]
+        # Use the last part in host IP 127.151.3.27 -> 27
+        # There can be no duplicates within the same test run
+        # thanks to how host registry registers subnets, and
```

Comment:
Can you specify it in `HostRegistry` (in the docstring at class definition)? Now it's a trick that depends on internal implementation detail of `HostRegistry`, instead it should be specified at the interface level.